### PR TITLE
remove bootstrap style import @import `'bootstrap/scss/custom-forms'`

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
### Problem statement
We depend on Bootstrap styles in our global styles a lot. But we don't plan to move forward with Bootstrap because we're working on our design system called Wildcard. To ease the transition to our styles and gain more control over visual changes, we want to move Bootstrap styles that we need into our codebase and eventually drop Bootstrap dependency.

### Refs
[Parent issue](https://github.com/sourcegraph/sourcegraph/issues/29200)
[SG issue](https://github.com/sourcegraph/sourcegraph/issues/30280)
[SG Issue 2](https://github.com/sourcegraph/sourcegraph/pull/32143)
[GitStart Task](https://app.gitstart.com/clients/sourcegraph/tickets/SG-30280)

## Preview App
[Storybook](https://client-sourcegraph-frontend-pr-220.onrender.com/)
[Frontend](https://client-sourcegraph-frontend-pr-220.onrender.com/)

### Success Criteria
- [ ] All highlighted imports has been removed
- [ ] All css rules currently used from the imports are now available in style sheets

### Implementation details

let's also remove the following imports and copy CSS rules that we need to our CSS files:

1. remove the import `@import 'bootstrap/scss/custom-forms` from global `index.scss` file
2. copy styles used in codebase into new global `custom-forms.scss` file within `client/branded/src/global-styles`
3. import file into global `index.scss`
Note: some style rules from @import 'bootstrap/scss/custom-forms are already being modified by styles in forms.scss

### ETA
70 minutes - initial implementation
180 minutes - with UI changes
 
## Test plan

Ensure the absence of UI regressions.

### CODE REVIEW

**Review by valerybugakov — COMMENTED** [review]
Let's rebase this PR from `main` after merging [this one](https://github.com/sourcegraph/sourcegraph/pull/30287). We need to remove the duplication of CSS variables that I mentioned in the linked PR review.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov requested change, for [this issue](https://github.com/sourcegraph/sourcegraph/issues/32117), has been added as part of this PR, kindly take a look

File: client/branded/src/global-styles/custom-forms.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
+:root {
+    --custom-select-padding-y: var(--input-padding-y);
+    --custom-select-padding-y-sm: var(--input-padding-y-sm);
+    --custom-select-padding-y-lg: var(--input-padding-y-lg);
+    --custom-select-padding-x: var(--input-padding-x);
+    --custom-select-padding-x-sm: var(--input-padding-x-sm);
+    --custom-select-padding-x-lg: var(--input-padding-x-lg);
+    --custom-select-font-size: 0.875rem;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/custom-forms.scss:8]: Should we depend on `--font-size-base` here instead?

File: client/branded/src/global-styles/variables.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+$input-font-size-sm: (11/14) + em;
+
+// Spacer
+$spacer: 1rem;
+
+:root {
+    --spacer: #{$spacer};
+    --font-size-base: #{$font-size-base};
+    --line-height-base: #{$line-height-base};
+    --input-font-size: var(--font-size-base);
```
Comment (valerybugakov) [inline client/branded/src/global-styles/variables.scss:16]: Let's group CSS variables by adding comments between groups like we do for SCSS variables.

```suggestion
    --line-height-base: #{$line-height-base};
    // Input
    --input-font-size: var(--font-size-base);
```

**Review by valerybugakov — CHANGES_REQUESTED** [review]

**Review by valerybugakov — APPROVED** [review]

**Pull request merged**